### PR TITLE
[anchors] Pluggable anchor commitments

### PR DESCRIPTION
**Pull request by halseth**

### CODE REVIEW

File: lnwallet/wallet.go [hunk 1]
```diff
@@ -747,24 +753,24 @@ func (l *LightningWallet) handleFundingCancelRequest(req *fundingReserveCancelMs
 // initial funding workflow as both sides must generate a signature for the
 // remote party's commitment transaction, and verify the signature for their
 // version of the commitment transaction.
-func CreateCommitmentTxns(localBalance, remoteBalance btcutil.Amount,
+func CreateCommitmentTxns(localBalance, remoteBalance,
```
Comment (joostjager) [inline lnwallet/wallet.go]: Shouldn't this complete function be moved into the plugin?

File: lnwallet/commitmenttx/commitment_type.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	WitnessScript []byte
+}
+
+// CommitmentType is a type that wraps the type of channel we are dealine with,
+// and abstracts the various ways of constructing commitment transactions. It
+// can be used to derive keys, scripts and outputs depending on the channel's
+// commitment type.
+type CommitmentType struct {
+	chanType channeldb.ChannelType
+}
```
Comment (joostjager) [inline lnwallet/commitmenttx/commitment_type.go]: I actually expected an interface here with three implementations?

File: contractcourt/chain_watcher.go [hunk 3]
```diff
@@ -331,7 +332,8 @@ func (c *chainWatcher) SubscribeChannelEvents() *ChainEventSubscription {
 // based off of only the set of outputs included.
 func isOurCommitment(localChanCfg, remoteChanCfg channeldb.ChannelConfig,
```
Comment (joostjager) [inline contractcourt/chain_watcher.go]: Shouldn't this be a method on `CommitmentType` too?

File: lnwallet/channel.go [hunk 4]
```diff
@@ -2375,7 +2304,7 @@ func (lc *LightningChannel) fundingTxIn() wire.TxIn {
 // createCommitmentTx generates the unsigned commitment transaction for a
 // commitment view and assigns to txn field.
 func (lc *LightningChannel) createCommitmentTx(c *commitment,
```
Comment (joostjager) [inline lnwallet/channel.go]: Not a method on the plugin?

**Comment by halseth** [comment]
Replaced by #3821

**Pull request closed without merging**